File: 0495_findPoisonedDuration.py
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        total = duration # last hit is added in advance
        for i in range(len(timeSeries) - 1):
            if timeSeries[i+1] - timeSeries[i] > duration:
                total += duration
            else:
                total += timeSeries[i+1] - timeSeries[i]
        return total

File: 0001-0500/0495_findPoisonedDuration.java
class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int totalDuration = 0;
        for (int i = 0; i < timeSeries.length; i++) {
            if (i != timeSeries.length - 1) {
                if (timeSeries[i+1] - timeSeries[i] < duration) {
                    totalDuration += timeSeries[i+1] - timeSeries[i];
                } else {
                    totalDuration += duration;
                }
            } else {
                totalDuration += duration;
            }
        }
        return totalDuration;
    }
}
